dev/TreeView/ViewModel.cpp

```diff
@@ -310,15 +318,85 @@ void ViewModel::NodeCollapsed(const winrt::event_token token)
 
 void ViewModel::SelectAll()
 {
+    BeginSelectionChanges();
+    auto trackSelection = gsl::finally([this]() { EndSelectionChanges(); });
```

Comment:
technically these should be reversed, if we throw during BeginSelectionChanges() and were called in a try block we would never clean up with EndSelectionChanges()

Reply:
Good point. Done.